### PR TITLE
remove strict slash req from amenities and states; add file storage test teardown

### PR DESCRIPTION
**Pull request by ZeroDayPoke**
image

### CODE REVIEW

File: api/v1/views/states.py [hunk 1]
```diff
@@ -26,20 +27,23 @@ def states_no_id_post():
...
     state_obj = validate_model("State", state_id)
     storage.delete(state_obj)
     return jsonify({})
 
-@app_views.route('/states/<state_id>', methods=['PUT'])
+
+@app_views.route('/states/<state_id>', methods=['PUT'], strict_slashes=False)
 def states_with_id_put(state_id=None):
     """states route handling - id given PUT scenario"""
     state_obj = validate_model("State", state_id)
```
Comment (Oggal) [inline api/v1/views/states.py:49]: Line 54 : We ignore the value 'state_id', this only needs to be ignored by cities.

Removing the unneeded ignore could improve runtime, though my larger concern is making sure the other classes are accurate as we move into those.

File: api/v1/views/amenities.py [hunk 2]
```diff
@@ -26,20 +27,25 @@ def amenities_no_id_post(amenity_id=None):
...
     storage.delete(amenity_obj)
     return jsonify({}), 200
 
-@app_views.route('/amenities/<amenity_id>', methods=['PUT'])
+
+@app_views.route('/amenities/<amenity_id>', methods=['PUT'],
+                 strict_slashes=False)
 def amenities_with_id_put(amenity_id=None):
     """amenities route - id given PUT scenario"""
     amenity_obj = validate_model('Amenity', amenity_id)
```
Comment (Oggal) [inline api/v1/views/amenities.py:51]: Line  56: We ignore the value 'state_id', this only needs to be ignored by cities.

Removing the unneeded ignore could improve runtime, though my larger concern is making sure the other classes are accurate as we move into those.
 (copied from state, but same issue)

File: api/v1/views/states.py (hunk 1 above)
Comment (ZeroDayPoke) [reply api/v1/views/states.py:49]: uwu; fixed

File: api/v1/views/amenities.py (hunk 2 above)
Comment (ZeroDayPoke) [reply api/v1/views/amenities.py:51]: uwu; fixed

File: tests/test_models/test_engine/test_file_storage.py [hunk 3]
```diff
@@ -71,6 +71,16 @@ def test_fs_func_docstrings(self):
...
+
+    @unittest.skipIf(models.storage_t == 'db', "not testing file storage")
+    @classmethod
+    def tearDownClass(cls):
+        """tears down test cls by removing jason :P after"""
+        try:
+            os.remove("file.json")
+        except IOError:
+            pass
+
```
[the remark this answers is not in the document]
Comment (ZeroDayPoke) [reply tests/test_models/test_engine/test_file_storage.py:83]: yea; hadn't looked into the FS testing inherited that much, but it appears to solve the issue

[the remark this answers is not in the document]
Comment (Oggal) [reply tests/test_models/test_engine/test_file_storage.py:83]: what issue is this solving?

[the remark this answers is not in the document]
Comment (ZeroDayPoke) [reply tests/test_models/test_engine/test_file_storage.py:83]: file storage mode discoverable test failure ideally; it worked locally

**Comment by Oggal** [comment]
Have you merged masters changes into your branch?

If so, let's run this against the main.py files from the checker and check the results. Assuming they pass we should be ready to merge this into master.

I think that should be part of our PR process going forward, to avoid needing a full PR for one or two missing checks

**Pull request merged**